### PR TITLE
feat/extract_speech

### PR DESCRIPTION
**Pull request by JarbasAl**
add support for VAD plugins to return only the speech from a audio chunk

### CODE REVIEW

File: ovos_plugin_manager/templates/vad.py [hunk 1]
```diff
@@ -45,6 +62,64 @@ def runtime_requirements(self):
...
+        """Generates audio frames from PCM audio data.
+        Takes the desired frame duration in milliseconds, the PCM data, and
+        the sample rate.
+        Yields Frames of the requested duration.
+        """
+        n = int(self.sample_rate * (self.frame_duration_ms / 1000.0) * 2)
+        offset = 0
+        timestamp = 0.0
+        duration = (float(n) / self.sample_rate) / 2.0
+
```
Comment (emphasize) [inline ovos_plugin_manager/templates/vad.py:75]: dont need `duration` here ( duration = self.frame_duration_ms / 1000.0)
But its late, am i wrong?

File: ovos_plugin_manager/templates/vad.py [hunk 2]
```diff
@@ -1,15 +1,32 @@
...
 from ovos_config import Configuration
 from ovos_utils import classproperty
 from ovos_utils.process_utils import RuntimeRequirements
 
 
+class Frame:
+    """Represents a "frame" of audio data."""
+
+    def __init__(self, bytes, timestamp, duration):
```
Comment (NeonDaniel) [inline ovos_plugin_manager/templates/vad.py:12]: Please add type annotations and docstrings to specify expected arg values

File: ovos_plugin_manager/templates/vad.py [hunk 3]
```diff
@@ -45,6 +62,64 @@ def runtime_requirements(self):
                                    no_internet_fallback=True,
                                    no_network_fallback=True)
 
+    def frame_generator(self, audio):
```
Comment (NeonDaniel) [inline ovos_plugin_manager/templates/vad.py:65]: Please add type annotation and/or update docstring to describe `audio`

File: ovos_plugin_manager/templates/vad.py [hunk 4]
```diff
@@ -45,6 +62,64 @@ def runtime_requirements(self):
...
+        offset = 0
+        timestamp = 0.0
+        duration = (float(n) / self.sample_rate) / 2.0
+
+        while offset + n <= len(audio):
+            yield Frame(audio[offset:offset + n], timestamp, duration)
+            timestamp += duration
+            offset += n
+
+    def extract_speech(self, audio):
```
Comment (NeonDaniel) [inline ovos_plugin_manager/templates/vad.py:81]: Please add type annotation and/or update docstring to describe audio

**Review by NeonDaniel — CHANGES_REQUESTED** [review]

**Pull request merged**